compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -14026,6 +14026,13 @@ TR::Register *OMR::Z::TreeEvaluator::inlineVectorBinaryOp(TR::Node *node, TR::Co
     TR_ASSERT_FATAL_WITH_NODE(node, node->getDataType().getVectorLength() == TR::VectorLength128,
         "Only 128-bit vectors are supported %s", node->getDataType().toString());
 
+    // Before z14, vector instructions for floating point operations supported only double-precision values.
+    // Starting with z14, single-precision (float) values are also supported.
+    TR_ASSERT_FATAL_WITH_NODE(node,
+        (node->getDataType().getVectorElementType() == TR::Float)
```

Comment:
Should not this whole condition be negated? Also, does it mean most of the Float vector test cases are going to hit that assert, if run on older than Z14 machine?